subprojects/common/result/src/main/kotlin/com/avito/android/Result.kt

```diff
@@ -106,7 +106,7 @@ sealed class Result<T> {
 
         override fun component1(): T? = null
 
-        override fun component2(): Throwable? = throwable
+        override fun component2(): Throwable = throwable
```

Comment:
Why so? parent function returns nullable

Reply:
If a compiler can infer a more precise type, let's do it.
This use case for Result is rare but valid. 

```kotlin
val result = Result.tryCatch<String> { throw RuntimeException() }
    
if (result is Result.Failure) {
    val (_, error: Throwable) = result
}
```